webtau-rest/src/test/groovy/com/twosigma/webtau/http/OpenAPISpecValidatorTest.groovy

```diff
@@ -0,0 +1,61 @@
+package com.twosigma.webtau.http
+
+import org.junit.Before
+import org.junit.Test
+
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder
+import static org.hamcrest.core.StringContains.containsString
+import static org.junit.Assert.assertThat
+
+class OpenAPISpecValidatorTest {
+    private final static String GET = "GET"
+    private final static String URL = "http://myhost.com:1234/"
+
+    private OpenAPISpecValidator validator
+
+    @Before
+    void setUp() throws Exception {
+        def specUrl = this.class.classLoader.getResource("test-spec.json")
+        validator = new OpenAPISpecValidator(specUrl.toString())
+    }
+
+    @Test
+    void "invalid responses should generate mismatches"() {
+        //This should generate two errors: one for missing mandatoryField and one for invalid type of intField.
+        //It should generate no errors for the missing optionalField
+        def testResponse = "{\"intField\": \"abc\"}"
+        def response = ok(testResponse)
+        def result = emptyResult()
+
+        validator.validateApiSpec(GET, URL, response, result)
+
+        result.mismatches.size().should == 2
+        assertThat(result.mismatches, containsInAnyOrder(
```

Comment:
oh boy.  Ok, let me try that.  Questionable in my view whether that's more readable than my version though 😄

Reply:
`~/regexp/` is a standard groovy syntax for regexps.
Main point here is ideally webtau matchers should be robust enough. if something is missing, it would be OK for a temp workaround using other lib. But long term I would prefer to improve core matchers of webtau. 
Since other libs won't work with `DataNode` out of the box, it is important to dog food webtau during normal tests as well. 
In long term, core matchers will be moved into a project of its own and be used for regular Unit Testing.

Reply:
Alright, you've convinced me, I'll switch it over 😄